### ISSUE TITLE
[$250] Over Limit Approver not seeing GBR or WS Chat unread when a report requires their approval

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#51689]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C06ML6X0W9L/p1729859922259099?thread_ts=1729596983.599409&cid=C06ML6X0W9L

## Action Performed:

1. Create Control workspace, invite Submitter, Approver A, Approver B
2. Set approval workflow in OldDot as Advanced Approval + Submitter > Approver A + Approver B as $10 overLimitForwardsTo
3. Confirm that Approver B has been invited to Submitters WS Chat by accessing the link to the WS chat (it shouldn't be unread or show up in LHN in focus mode)
4. As Submitter, create expense for + $10
5. As Approver B, click on the chat and then click away to another chat to make sure the WS chat is read (not bold) in your LHN
6. As Approver A, approve the expense

## Expected Result:
Approver B should see that the WS Chat is GBR and unread.

## Actual Result:
The WS Chat isn't unread for Approver B and there is no GBR. Once Approver B accesses the WS Chat GBR shows and persists.

## Workaround:
No

## Platforms:
All

## Screenshots/Videos

https://github.com/user-attachments/assets/1be34913-9f5f-4fed-a4ff-1e5dc6335650

 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021852100736496847321</li>
        <li>Upwork Job ID: 1852100736496847321</li>
        <li>Last Price Increase: 2024-10-31</li>
    </ul>
</details>

**Comment by Beamanator** [comment]
This looks pretty fun & internal 👍

**Comment by Beamanator** [comment]
Finding some VERY interesting things while workin on this :D

**Comment by Beamanator** [comment]
Ready for review finally!

**Comment by Beamanator** [comment]
Hmm it looks like even submitting a report causes `overLimitForwardsTo` to get the workspace chat unread, i think we have a bug for that somewhere right @garrettmknight ?

**Comment by Beamanator** [comment]
PR merged! Hopefully this is working well after tomorrow's deploy - not the "unread" changes yet, just GBR for now

**Comment by Beamanator** [comment]
PR was deployed 30 minutes ago! @garrettmknight the GBR part is hopefully working perfectly now 🙏 

I knew the Unread part wasn't handled, but I'm wondering it that will go on hold just as https://github.com/Expensify/App/issues/51723 just went on hold?

**Comment by Beamanator** [comment]
One more comment just to make this go off overdue

**Comment by garrettmknight** [comment]
> I knew the Unread part wasn't handled, but I'm wondering it that will go on hold just as https://github.com/Expensify/App/issues/51723 just went on hold?

Great call. Once the GBR is complete here, I think we can close and let the other issue handle itself.

**Comment by Beamanator** [comment]
Noice 🙏  then this looks ready to re-test! @johncschuster would you mind doing the retest? & maaaaaaaybe also doing the regressions tests? Or i could do the regression tests, but i guess we can unassign a C+ here?

**Comment by Beamanator** [comment]
Oh I guess we could get Applause to test this too, before closing this out

Proposed regression steps:

Normal report forwarding (do this in both newdot & olddot):
1. Create Control policy w/ advanced approval mode & NOT instant submit & has policy chats enabled
2. Set up approvals like: Submitter `submitsTo` Approver 1 who `forwardsTo` Approver 2
3. As submitter, Create expense, submit to approver 1
4. Log in to approver 2 on a device so you can see live onyx updates when you do the next step
5. As approver 1, approve the report (this will forward it to approver 2)
6. Verify in Approver 2's account, without opening any new reports, the expense report live-updates w/ GBR

Report forwarding w/ `overLimitForwardsTo`:
1. Same step 1 as above (you can use the same policy)
2. Set up an approval like so:
    1. Submitter `submitsTo` Approver 1
    2. Approver 1 has `overLimitForwardsTo` set to Approver 2, with `approvalLimit` set to something like `$10.00`
3. As submitter, Create expense for $11.00, submit to approver 1
4. Log in to approver 2 on a device so you can see live onyx updates when you do the next step
5. As approver 1, approve the report (this will forward to approver 2)
6. Verify in Approver 2's account, without opening any new reports, the expense report live-updates w/ GBR

**Comment by Beamanator** [comment]
@johncschuster @garrettmknight can one of y'all do a retest OR ask applause? Or if you need me to retest let me know

**Comment by johncschuster** [comment]
@Beamanator sorry I missed your comment above! Sure, I'll get a retest now!

**Comment by izarutskaya** [comment]
For me WS Chat isn't unread for Approver B. Check please my video

https://github.com/user-attachments/assets/9ac1a293-fdc2-4375-a3d4-e59d910eb22f

**Comment by Beamanator** [comment]
@izarutskaya thanks for testing, but we actually don't need to worry about the "Unread" state anymore

Tests for this issue should be here: https://github.com/Expensify/App/issues/51689#issuecomment-2464267297

We agreed [here](https://github.com/Expensify/App/issues/51689#issuecomment-2460048226)  that in this issue we'll only worry about the GBR, not unreads

**Comment by johncschuster** [comment]
@izarutskaya bump on the above comment!

**Comment by Beamanator** [comment]
Waiting for 1 final test 🤣

**Comment by Beamanator** [comment]
I believe we can close this, since the only thing not working from here - https://github.com/Expensify/App/issues/51689#issuecomment-2482881647 - is the "Unreads" which we agreed isn't going to be handled in this issue
